src/ast.rkt

```diff
@@ -17,7 +21,8 @@
 (struct vec-decl (id size) #:transparent)
 
 ; Set externally-declared vector in memory, must be loaded from.
-(struct vec-extern-decl (id size) #:transparent)
+; Tag optionally says if this is an input or an output.
+(struct vec-extern-decl (id size tag) #:transparent)
```

Comment:
It doesn't look like the tag is optional?

Reply:
Updated comment.